script/split_tests.py

```diff
@@ -0,0 +1,197 @@
+            self._current_bucket.add(tests)
+            return
+
+        if isinstance(tests, TestFolder):
+            for test in tests.children.values():
+                self.split_tests(test)
+            return
+
+        # Create new bucket
+        if len(self._buckets) == self._bucket_count:
+            # Last bucket, add all tests to it
+            self._last_bucket = True
+        else:
+            self._current_bucket = Bucket()
+            self._buckets.append(self._current_bucket)
+
+        # Add test to new bucket
+        self.split_tests(tests)
+
+    def create_ouput_file(self) -> None:
+        """Create output file."""
+        with open("pytest_buckets.txt", "w") as file:
+            for bucket in self._buckets:
+                print(f"Bucket has {bucket.tests} tests")
+                file.write(bucket.get_paths_line())
+
+
+@dataclass
+class TestFile:
+    """Class to hold number of tests."""
+
+    path: str
+    total_tests: int
+
+    def __gt__(self, other):
+        """Return if greater than."""
+        return self.total_tests > other.total_tests
+
+
+@dataclass
+class TestFolder:
+    """Class to hold test information."""
+
+    path: str
+    children: dict[str, TestFolder | TestFile] = field(default_factory=dict)
+
+    @property
+    def total_tests(self) -> int:
+        """Return total tests."""
+        return sum([test.total_tests for test in self.children.values()])
+
+    def __repr__(self):
+        """Return representation."""
+        return f"TestFolder(total={self.total_tests}, children={len(self.children)})"
+
+
+def insert_at_correct_position(
+    test_holder: TestFolder, test_path: str, total_tests: int
+) -> None:
+    """Insert test at correct position."""
+    current_path = test_holder
+    for part in test_path.split("/")[1:]:
+        if part.endswith(".py"):
+            current_path.children[part] = TestFile(test_path, total_tests)
+        else:
+            current_path = current_path.children.setdefault(
+                part, TestFolder(os.path.join(current_path.path, part))
+            )
+
+
+def collect_tests(path: str) -> tuple[TestFolder, TestFile]:
+    """Collect all tests."""
+    result = subprocess.run(
+        ["pytest", "--collect-only", "-qq", "-p", "no:warnings", path],
+        check=False,
+        capture_output=True,
+        text=True,
+    )
+
+    if result.returncode != 0:
+        print("Failed to collect tests:")
+        print(result.stderr)
+        print(result.stdout)
+        sys.exit(1)
+
+    folder = TestFolder(path.split("/")[0])
+    insert_at_correct_position(folder, path, 0)
+    max_tests_in_file = TestFile("", 0)
+
+    for line in result.stdout.splitlines():
+        if not line.strip():
+            continue
+        parts = [x.strip() for x in line.split(":")]
+        if len(parts) != 2:
+            print(f"Unexpected line: {line}")
+            sys.exit(1)
+
+        path = parts[0]
+        total_tests = int(parts[1])
```

Comment:
`line.partition(":")` is safer. It will always return 3 parts.

```suggestion
        path, _, total_tests = line.partition(": ")
        if not path or not total_tests:
            print(f"Unexpected line: {line}")
            sys.exit(1)

        total_tests = int(total_tests)
```